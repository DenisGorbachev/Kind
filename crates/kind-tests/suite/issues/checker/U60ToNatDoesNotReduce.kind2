Doit3times <x> (f: (x -> x)) (n: x) : x
Doit3times f n = (f (f (f n)))

#partial
Nat.zero : (Nat)

U60.to_nat (x: U60) : (Nat)
U60.to_nat 0 = (Nat.zero)
U60.to_nat n = (Nat.succ (U60.to_nat (- n 1)))

#partial
Nat.add (a: (Nat)) (b: (Nat)) : (Nat)
Nat.add (Nat.succ a) b = (Nat.succ (Nat.add a b))
Nat.add (Nat.zero) b = b

#partial
Nat.succ (pred: (Nat)) : (Nat)

#axiom
Nat : Type

Main : _
Main = (let a = (Doit3times ((x => (Nat.mul x x)) :: Nat -> Nat) (U60.to_nat 2)); a)

#partial
Nat.mul (a: (Nat)) (b: (Nat)) : (Nat)
Nat.mul (Nat.succ a) b = (Nat.add (Nat.mul a b) b)
Nat.mul (Nat.zero) b = (Nat.zero)

Test_anon_fun : (Equal  (Nat.succ (Nat.succ (Nat.zero))) (U60.to_nat 5))
Test_anon_fun = ?

#axiom
Equal <t> (a: t) (b: t) : Type
