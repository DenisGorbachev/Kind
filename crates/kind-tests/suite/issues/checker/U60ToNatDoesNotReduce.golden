   INFO  Inspection

      * Hole: (Equal _ 2n 5n) 


      /--[suite/issues/checker/U60ToNatDoesNotReduce.kind2:30:17]
      |
   29 |    Test_anon_fun : (Equal  (Data.Nat.succ (Data.Nat.succ (Data.Nat.zero))) (Data.U60.to_nat 5))
   30 |    Test_anon_fun = ?
      |                    v
      |                    \Here!
   31 |    


